experimental/dds/tree2/src/feature-libraries/editable-tree-2/lazyField.ts

```diff
@@ -202,6 +205,76 @@ export abstract class LazyField<TKind extends FieldKindTypes, TTypes extends All
 		);
 	}
 
+	public reduce<U>(
+		callbackfn: (
+			previousValue: U,
+			currentValue: UnboxNodeUnion<TTypes>,
+			currentIndex: number,
+			sequence: this,
+		) => U | Stop<U>,
+		initialValue?: U | undefined,
+	): UnboxNodeUnion<TTypes> | U {
+		return this.reduceImpl(
+			callbackfn,
+			(cursor) => unboxedUnion(this.context, this.schema, cursor),
+			initialValue,
+		);
+	}
+
+	public reduceBoxed<U>(
+		callbackfn: (
+			previousValue: U,
+			currentValue: TypedNodeUnion<TTypes>,
+			currentIndex: number,
+			sequence: this,
+		) => U | Stop<U>,
+		initialValue?: U | undefined,
+	): TypedNodeUnion<TTypes> | U {
+		return this.reduceImpl(
+			callbackfn,
+			(cursor) => makeTree(this.context, cursor) as TypedNodeUnion<TTypes>,
+			initialValue,
+		);
+	}
+
+	private reduceImpl<Accumulation, T>(
+		callbackfn: (
+			previousValue: Accumulation,
+			currentValue: T,
+			currentIndex: number,
+			sequence: this,
+		) => Accumulation | Stop<Accumulation>,
```

Comment:
The addition of 'Stop' is clever and can be done in a non-breaking way, but we should wait until we have compatibility tests before we tinker with enhancements.